File: golang/compare-with-py/add.py
import time
import os

print("Python")

upper_limit = int(os.getenv("UPPER_LIMIT"))
if upper_limit == 0:
    raise ValueError("no limit")
start_time = time.time()

data = []
for i in range(upper_limit):
    data.append(i)

res = 0
for d in data:
    res += d

end_time = time.time()
print("Result:", res)
print("Execution time:", (end_time - start_time) * 1_000_000, "µs")

File: golang/compare-with-py/add.go
package main

import (
	"fmt"
	"time"
)

func main() {
	fmt.Println("Golang")
	startTime := time.Now()
	data := []uint64{} //no preallocation to make it harder for go
	for i := 0; i < 1000; i++ {
		data = append(data, uint64(i))
	}
	var res uint64 = 0
	for _, d := range data {
		res += d * 2
	}
	endTime := time.Now()
	fmt.Println("Result: ", res)
	fmt.Println("Execution time:", endTime.Sub(startTime))
}
